### ISSUE TITLE
When the Cow Comes Home (Quest. Some problems with Bessy NPC).

### ISSUE THREAD

**Issue by Rorschach91** [azerothcore/azerothcore-wotlk#19584]
Triager Notes:

Original CC Report: https://github.com/chromiecraft/chromiecraft/issues/5233#issue-1634065525

Unfortunately, I have not been able to find any videos that I can consider 100% reliable.

In any case i've some evidence by Wowhead comments that seems confirming this report:

![prova 3](https://github.com/user-attachments/assets/86309833-bf37-4caa-880d-bf9ea9f5bf1f)
![prova 4](https://github.com/user-attachments/assets/6c50aba6-4678-488b-9414-367979ce76f7)
![prova 6](https://github.com/user-attachments/assets/28c0ee05-27a2-45df-80f7-68d11bee162b)

And also some comments that confirming this quest also had problems at TBC times:

![Prova 1](https://github.com/user-attachments/assets/01477891-d20b-44c4-8f13-a099630cf7a0)
![prova 2](https://github.com/user-attachments/assets/9d873564-9c34-4d3c-b1ce-a4fdf11b15b7)

My final considerations and other bugs:

Considering what I've read and what i've seen: 

1. Bessy should attack hostile npcs, but it doesn't.

2. Seems that Bessy hasn't an attack animation in the official TBC.

3. My log shows no attacks by Bessy against hostile npcs.

4. If you are fighting, Bessy go on its way without waiting you. It should not do it.

5. If Bessy gets hit by Frost Nova ability (hostile npcs mage) It stops going on, even when Frost Nova effect fades.
    This last bug forces the player to restart the quest.

### What client do you play on?

enUS

### Faction

Both

### Content Phase:

70

### Current Behaviour

Bessy completely ignores any NPCs during the escort

### Expected Blizzlike Behaviour

In the history of WOW, Escort NPC never ignores Enemy NPC, I hope this is enough.

### Source

No such thing as ignore NPCs

### Steps to reproduce the problem

accept When the Cow Comes home
Observe Bessy's behaviour
end

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/40ff0c607ecec81e9793d521cc1adfe3883e20ab

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna](https://github.com/azerothcore/mod-eluna)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [mod-instanced-worldbosses](https://github.com/nyeriah/mod-instanced-worldbosses)
- [mod-zone-difficulty](https://github.com/azerothcore/mod-zone-difficulty)
- [mod-BreakingNewsOverride](https://github.com/AnchyDev/BreakingNewsOverride)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-event-scripts](https://github.com/55Honey/Acore_eventScripts)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Customizations

None

### Server

ChromieCraft

**Comment by heyitsbench** [comment]
Based on my experience on TBC/Wrath Classic, Bessy should have a defensive react state, i.e., only attacking those who attack her first, she should not attack players that aggro onto and only attack the player character.

**Comment by Rorschach91** [comment]
Yep, but in this case It does not even attack npcs that attack it.
